credentials/src/main/kotlin/web5/sdk/credentials/StatusListCredential.kt

```diff
@@ -0,0 +1,345 @@
+      "Status purpose in the credential to validate is null"
+    }
+
+    require(statusListCredStatusPurpose != null) {
+      "Status purpose in the status list credential is null"
+    }
+
+    require(statusListEntryValue.statusPurpose == statusListCredStatusPurpose) {
+      "Status purposes do not match between the credentials"
+    }
+
+    val compressedBitstring: String? =
+      statusListCredential.vcDataModel.credentialSubject.jsonObject[ENCODED_LIST] as? String?
+
+    require(!compressedBitstring.isNullOrEmpty()) {
+      "Compressed bitstring is null or empty"
+    }
+
+    val credentialIndex = statusListEntryValue.statusListIndex
+    val expandedValues: List<String> = bitstringExpansion(compressedBitstring)
+
+    return expandedValues.any { it == credentialIndex }
+  }
+
+
+  /**
+   * Validates if a given credential is part of the status list.
+   *
+   * @param credentialToValidate The [VerifiableCredential] to be validated against the status list.
+   * @param httpClient An optional [HttpClient] for fetching the status list credential. If not provided, a default HTTP client will be used.
+   * @return A [Boolean] indicating whether the `credentialToValidate` is part of the status list.
+   *
+   * This function fetches the status list credential from a URL present in the `credentialToValidate`.
+   * It supports using either a user-provided `httpClient` or a default client when no client is passed in.
+   * The function then checks if the given `credentialToValidate`'s status list index is present in the expanded status list derived from the fetched status list credential.
+   *
+   * Example:
+   * ```
+   * val isRevoked = validateCredentialInStatusList(credentialToCheck)
+   * ```
+   */
+  public fun validateCredentialInStatusList(
+    credentialToValidate: VerifiableCredential,
+    httpClient: HttpClient? = null // default HTTP client but can be overridden
+  ): Boolean {
+    return runBlocking {
+      var isDefaultClient = false
+      val clientToUse = httpClient ?: defaultHttpClient().also { isDefaultClient = true }
+
+      try {
+        val statusListEntryValue: StatusList2021Entry =
+          StatusList2021Entry.fromJsonObject(credentialToValidate.vcDataModel.credentialStatus.jsonObject)
+        val statusListCredential =
+          clientToUse.fetchStatusListCredential(statusListEntryValue.statusListCredential.toString())
+
+        return@runBlocking validateCredentialInStatusList(credentialToValidate, statusListCredential)
+      } finally {
+        if (isDefaultClient) {
+          clientToUse.close()
+        }
+      }
+    }
+  }
+
+  private fun defaultHttpClient(): HttpClient {
+    return HttpClient(CIO) {
+      install(ContentNegotiation) {
+        jackson { jacksonObjectMapper() }
+      }
+    }
+  }
+
+  private suspend fun HttpClient.fetchStatusListCredential(url: String): VerifiableCredential {
+    try {
+      val response: io.ktor.client.statement.HttpResponse = this.get(url)
```

Comment:
Could this be imported? 
```suggestion
      val response: HttpResponse = this.get(url)
```